client/init.c

```diff
@@ -189,6 +189,8 @@ int xdag_init(int argc, char **argv, int isGui)
 		return -1;
 	}
 
+	g_xdag_auto_pick_pool = !mining_threads_count ? 1 : (!is_pool && pool_arg == NULL); /* auto pick pool */
```

Comment:
It seems g_xdag_auto_pick_pool  is always 1.
xdag  is started as a pool - mining_threads_count is 0 and g_xdag_auto_pick_pool  = 1
xdag is started as a wallet without mining - mining_threads_count is 0 and g_xdag_auto_pick_pool = 1
xdag is started as a wallet with mining - mining_threads_count is > 0 but (!is_pool && pool_arg == NULL)  is true
And why do you need (!is_pool && pool_arg == NULL)?